fs/readdircount_linux_amd64.go

```diff
@@ -0,0 +1,67 @@
+// +build linux,amd64
+
+package fs
+
+import (
+	"errors"
+	"os"
+	"syscall"
+	"unsafe"
+)
+
+func countDirEntries(buf []byte, n int) int {
+	count := 0
+	buf = buf[:n]
+	for len(buf) > 0 {
+		// see man page getdents(2) for struct linux_dirent64
+		reclen := *(*uint16)(unsafe.Pointer(&buf[16]))
```

Comment:
In amd64, `syscall.Dirent` maps to `struct linux_dirent64`, see https://golang.org/src/syscall/ztypes_linux_amd64.go?s=1903:2035#L124